core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -1355,6 +1392,20 @@ public void removeAll() {
     }
   }
   
+  private void markAllCacheEventsFireable() {
+    while (!EventThreadLocal.get().isEmpty()) {
```

Comment:
I do not like the use of static here. Everything could be hidden behind the `EventNotificationService` api.